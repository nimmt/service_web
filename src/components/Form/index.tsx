import React from 'react';

interface IFormProps {
  onSubmit(data: any): any;
}

interface ITextInputProps {
  name: string;
  register: any;
  required?: boolean;
}

const Form: React.FC<IFormProps> = ({ onSubmit, children }) => (
  <form onSubmit={onSubmit}>
    {children}
  </form>
);

export const TextInput: React.FC<ITextInputProps> = ({
  name,
  register,
  required = false
}) => (
  <input
    type="text"
    name={name}
    ref={register({
      required: required
    })}
  />
);

export default Form;
