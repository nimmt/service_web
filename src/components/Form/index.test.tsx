import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Form, { TextInput } from './index';

it('renders correctly', () => {
  const tree = ReactTestRenderer
    .create(
      <Form onSubmit={jest.fn()}>
        <TextInput
          name="name"
          register={jest.fn()}
          required
        />
      </Form>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
});
