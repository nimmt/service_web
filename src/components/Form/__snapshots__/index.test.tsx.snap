// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    name="name"
    type="text"
  />
</form>
`;
